x/group/types_test.go

```diff
@@ -0,0 +1,137 @@
+package group_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/cosmos/cosmos-sdk/x/group/internal/math"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAllow(t *testing.T) {
+	testCases := []struct {
+		name           string
+		policy         *group.PercentageDecisionPolicy
+		tally          *group.Tally
+		totalPower     string
+		votingDuration time.Duration
```

Comment:
```suggestion
		votingDuration time.Duration
		result DecisionPolicyResult
```

I believe there are too many many logic in this test, which makes it hard to read. Could we refactor to add a result in the expected test case, and simply do:

```
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {

			policyResult, err := tc.policy.Allow(*tc.tally, tc.totalPower, tc.votingDuration)
			require.NoError(t, err)
			require.Equal(t, tc.result, policyResult)
}
```